kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/source/CassandraTableReader.scala

```diff
@@ -213,65 +213,60 @@ class CassandraTableReader(private val name: String,
     *
     * @param future Cassandra Future ResultSet to iterate over.
     */
-  private def process(future: Future[ResultSet]) = {
+  private def process(future: Future[ResultSet]): Unit = {
     //get the max offset per query
     var maxOffset: Option[String] = None
     //on success start writing the row to the queue
-    future.onSuccess({
+    future.onSuccess {
       case rs: ResultSet =>
-        logger.info(s"Connector $name processing results for $keySpace.$table.")
-        val iter = rs.iterator()
-        var counter = 0
-
-        var index = 100
-        var shouldStop = stop.get()
-        // process results unless told to stop
-        while (iter.hasNext & !shouldStop) {
-          //buffer the value to avoid going to memory everytime via the volatile variable
-          index -= 1
-          if (index == 0) {
-            index = 100
-            shouldStop = stop.get()
-          }
-
-          // this is asynchronous
-          if ((rs.getAvailableWithoutFetching == setting.fetchSize / 2) && !rs.isFullyFetched) rs.fetchMoreResults
-
-          val row = iter.next()
-          Try {
-            // if not bulk get the maxOffset value 
-            if (!bulk) {
-              maxOffset = if (isTokenBased) {
-                getTokenMaxOffsetForRow(maxOffset, row)
-              } else {
-                getTimebasedMaxOffsetForRow(maxOffset, row)
+        try {
+          logger.info(s"Connector $name processing results for $keySpace.$table.")
+          val iter = rs.iterator()
+          var counter = 0
+
+          // process results unless told to stop
+          while (iter.hasNext & !stop.get()) {
+            // this is asynchronous
+            if ((rs.getAvailableWithoutFetching == setting.fetchSize / 2) && !rs.isFullyFetched) rs.fetchMoreResults
+
+            val row = iter.next()
+            Try {
+              // if not bulk get the maxOffset value
+              if (!bulk) {
+                maxOffset = if (isTokenBased) {
+                  getTokenMaxOffsetForRow(maxOffset, row)
+                } else {
+                  getTimebasedMaxOffsetForRow(maxOffset, row)
+                }
+                logger.debug(s"Connector $name max Offset is currently: ${maxOffset.get}")
+              }
+              if(processRow(row)) {
+                counter += 1
               }
-              logger.debug(s"Connector $name max Offset is currently: ${maxOffset.get}")
+            } match {
+              case Failure(e) =>
+                logger.error(s"Connector $name error processing row ${row.toString} for table $keySpace.$table.", e)
+                throw new ConnectException(s"Connector $name error processing row ${row.toString} for table $keySpace.$table.", e)
```

Comment:
trying to follow code flow
is the `case Failure(e)` catching the exception from `processRow` but then throwing a `ConnectException`  that we recatch below?

Reply:
nice catch. 
i just cleaned that code. The Try block was not adding any value.

what we don't do is propagate the exception to connect to retry. after this PR maybe we address that.